File: content/polympiads/3.temps-pour-rejoindre-tous-les-arrets/codes/solution.cpp
#include <vector>
#include <iostream>

using namespace std;

int main () {
    int nbArrets;
    cin >> nbArrets;
    vector<int> durees(nbArrets - 1);
    for (int i = 0; i < durees.size(); i ++) {
        cin >> durees[i];
    }

    // Pour chaque fin de trajet
	for (int iFin = 1; iFin < nbArrets; iFin ++) {
		// On calcule le temps de trajet à la fin
		int tempsAFin = 0;
		for (int iArret = 0; iArret < iFin; iArret ++) {
			tempsAFin += durees[iArret];
		}
		
		// On affiche le temps
	    cout << tempsAFin << endl;
	}

    // BONUS
    
    // Pour chaque fin de trajet
    int tempsAFin = 0;
	for (int iFin = 1; iFin < nbArrets; iFin ++) {
		// On augmente le temps par rapport au temps précédent (voir les indices)
		tempsAFin += durees[iFin - 1];
		
		// On affiche le temps
	    cout << tempsAFin << endl;
	}
}

File: content/polympiads/3.temps-pour-rejoindre-tous-les-arrets/codes/solution.py
nb_arrets = int(input())
durees = list(map(int, input().split()))

# Pour chaque fin
for i_fin in range(1, nb_arrets - 1):
    # On calcule le temps de trajet à la fin
    temps_a_fin = 0
    for i_arret in range(i_fin): temps_a_fin += durees[i_arret]
    
    # On affiche le temps
    print(temps_a_fin)


# BONUS

nb_arrets = int(input())
durees = list(map(int, input().split()))

temps_a_fin = 0

# Pour chaque fin
for i_fin in range(1, nb_arrets - 1):
    # On augmente le temps par rapport au temps précédent (voir les indices)
    temps_a_fin += durees[i_fin - 1]

    # On affiche le temps
    print(temps_a_fin)